app.go

```diff
@@ -66,79 +65,105 @@ func (a *App) getAlerts(w http.ResponseWriter, r *http.Request) {
+	Comment   string    `json:"comment" schema:"Comment"`
+	CreatedBy string    `json:"createdBy" schema:"CreatedBy"`
+	Matchers  []Matcher `json:"matchers" schema:"Matchers"`
+	Schedule  Schedule  `json:"schedule" schema:"Schedule"`
+}
+
+type Matcher struct {
+	Name    string `json:"name" schema:"Name"`
+	Value   string `json:"value" schema:"Value"`
+	IsRegex bool   `json:"isRegex" schema:"IsRegex"`
 }
 
 // Schedule structure
 type Schedule struct {
-	StartTime string `json:"start_time"`
-	EndTime   string `json:"end_time"`
-	Repeat    Repeat `json:"repeat"`
+	StartTime string `json:"start_time" schema:"StartTime"`
+	EndTime   string `json:"end_time" schema:"EndTime"`
+	Repeat    Repeat `json:"repeat" schema:"Repeat"`
 }
 
 // Repeat structure
 type Repeat struct {
-	Enabled  bool   `json:"enabled"`
-	Interval string `json:"interval"`
-	Count    int    `json:"count"`
+	Enabled  bool   `json:"enabled" schema:"-"`
+	Interval string `json:"interval" schema:"Interval"`
+	Count    int    `json:"count" schema:"Count"`
 }
 
 // Valid validates a silence request
-func (r APISilenceRequest) Valid() bool {
-	if r.Comment == "" || r.CreatedBy == "" {
-		return false
+func (r APISilenceRequest) Valid() (string, bool) {
+	if r.Comment == "" {
+		return "comment field empty", false
 	}
+
+	if r.CreatedBy == "" {
+		return "createdBy field empty", false
+	}
+
 	if len(r.Matchers) < 1 {
-		return false
+		return "number of matchers should be bigger than 0", false
 	}
 
-	err := r.Matchers.Validate(strfmt.Default)
-	if err != nil {
-		return false
+	for _, m := range r.Matchers {
+		if !m.Valid() {
+			return fmt.Sprintf("matcher '%v' is invalid", m), false
+		}
 	}
 
-	if r.Schedule.Valid() != true {
-		return false
+	msg, ok := r.Schedule.Valid()
+	if !ok {
+		return msg, false
 	}
 
-	if r.Schedule.Repeat.Valid() != true {
-		return false
+	msg, ok = r.Schedule.Repeat.Valid()
+	if !ok {
+		return msg, false
 	}
+	return "", true
+}
+
+func (m Matcher) Valid() bool {
 	return true
 }
 
-//Valid returns true if the schedule is valid
-func (s Schedule) Valid() bool {
+// Valid returns true if the schedule is valid
+func (s Schedule) Valid() (string, bool) {
 	if s == (Schedule{}) {
-		return false
+		return "empty schedule provided", false
 	}
+
 	_, err := time.Parse(requestTimeLayout, s.StartTime)
 	if err != nil {
-		return false
+		return "invalid start time format", false
 	}
+
 	_, err = time.Parse(requestTimeLayout, s.EndTime)
 	if err != nil {
-		return false
+		return "invalid end time format", false
 	}
-	return true
+	return "", true
 }
 
 //Valid returns true if the repeat is valid
-func (r Repeat) Valid() bool {
+func (r Repeat) Valid() (string, bool) {
 	if r == (Repeat{}) {
-		return false
+		return "schedule repeat is empty", false
+	}
+
+	if r.Count <= scheduleCountMin {
+		return fmt.Sprintf("repeat count must be higher than %d", scheduleCountMin), false
 	}
-	if r.Count > 50 {
-		return false
+
+	if r.Count >= scheduleCountMax {
```

Comment:
As per returned text:
```suggestion
	if r.Count > scheduleCountMax {
```

Reply:
changed